examples/unsat_root_message_no_version.rs

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use pubgrub::error::PubGrubError;
+use pubgrub::range::Range;
+use pubgrub::report::Reporter;
+use pubgrub::solver::{resolve, OfflineDependencyProvider};
+use pubgrub::version::SemanticVersion;
+
+use pubgrub::report::{DefaultStringReporter, External, ReportFormatter};
+use pubgrub::term::Term;
+use pubgrub::type_aliases::Map;
+use std::fmt::{self, Display};
+
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
+pub enum Package {
+    Root,
+    Package(String),
+}
+
+impl Display for Package {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Package::Root => write!(f, "root"),
+            Package::Package(name) => write!(f, "{}", name),
+        }
+    }
+}
+
+#[derive(Debug, Default)]
+struct CustomReportFormatter;
+
+impl ReportFormatter<Package, Range<SemanticVersion>> for CustomReportFormatter {
+    type Output = String;
+
+    fn format_terms(&self, terms: &Map<Package, Term<Range<SemanticVersion>>>) -> String {
+        let terms_vec: Vec<_> = terms.iter().collect();
+        match terms_vec.as_slice() {
+            [] => "version solving failed".into(),
+            // TODO: special case when that unique package is root.
```

Comment:
I think that comment is not relevant anymore since you did special-case root.

Reply:
```suggestion
```